src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,454 @@
+   * 'allow-crossing' - allows thumb to cross other thumbs. Default.
+   * 'inhibit-crossing'- keeps the thumb from crossing and separated by a step.
+   * @default 'inhibit-crossing'
+   */
+  thumbMode?: 'allow-crossing' | 'inhibit-crossing';
+  /**
+   * Additional props to include on each thumb container.
+   */
+  thumbProps?: Omit<CommonProps, 'title'>;
+  /**
+   * Callback fired when the value(s) of the slider has changed. This will receive
+   * changes at the end of a slide as well as changes from clicks on rails and tracks.
+   */
+  onChange?: (values: ReadonlyArray<number>) => void;
+  /**
+   * Callback fired when the value(s) of the slider are internally updated during
+   * operations like dragging a Thumb. Use this callback with caution as a
+   * high-volume of updates will occur when dragging.
+   */
+  onUpdate?: (values: ReadonlyArray<number>) => void;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * Slider component that display Thumbs for each value specified along a Rail.
+ * @example
+ * <Slider values={[10]} min={0} max={60} disabled />
+ * <Slider values={[10, 20]} min={0} max={50} step={2} />
+ * <Slider values={[10, 20, 30, 40]} min={0} max={60} setFocus
+ *   thumbMode='allow-crossing' />
+ */
+export const Slider = React.forwardRef<HTMLDivElement, SliderProps>(
+  (props, ref) => {
+    const {
+      min = 0,
+      max = 100,
+      values,
+      step = 1,
+      setFocus = false,
+      tooltipProps,
+      tooltipRenderer,
+      disabled = false,
+      tickLabels,
+      minLabel,
+      maxLabel,
+      trackDisplayMode = undefined,
+      thumbMode = 'inhibit-crossing',
+      onChange,
+      onUpdate,
+      thumbProps,
+      className,
+      ...rest
+    } = props;
+
+    const [currentValues, setCurrentValues] = React.useState(values);
+    React.useEffect(() => {
+      setCurrentValues(values);
+    }, [values]);
+
+    const [minValueLabel, setMinValueLabel] = React.useState(minLabel);
+    React.useEffect(() => {
+      setMinValueLabel(minLabel ?? min.toString());
+    }, [minLabel, min]);
+
+    const [maxValueLabel, setMaxValueLabel] = React.useState(maxLabel);
+    React.useEffect(() => {
+      setMaxValueLabel(maxLabel ?? max.toString());
+    }, [maxLabel, max]);
+
+    const [trackDisplay, setTrackDisplay] = React.useState<TrackDisplayMode>(
+      () => getDefaultTrackDisplay(trackDisplayMode, currentValues),
+    );
+    React.useEffect(() => {
+      setTrackDisplay(getDefaultTrackDisplay(trackDisplayMode, currentValues));
+    }, [trackDisplayMode, currentValues]);
+
+    useTheme();
+    const containerRef = React.useRef<HTMLDivElement>(null);
+
+    React.useEffect(() => {
+      if (containerRef.current && setFocus) {
+        const firstThumb = containerRef.current.querySelector(
+          '.iui-slider-thumb',
+        );
+        firstThumb && (firstThumb as HTMLElement).focus();
+      }
+    }, [setFocus]);
+
+    const fractionDigits = React.useMemo(() => {
```

Comment:
The function name is not really right.. It does not return 0, if the number passed is integer..
Also, it is used only in `formatNumberValue` where that check is done.. Maybe just move this one to that util function? You pass there same `step`..No need to change function ref whenever step is changed..

Reply:
I changed the name of the callback function, but I left it as a useMemo to avoid having to convert the step to a string and look up the '.' each time I want to generate the text for the tooltip. The tooltip get generated a lot the step value rarely changes.

Reply:
well.. discussable, but not gonna be picky, cause you will never contribute again :DD